core/src/main/scala/org/http4s/Headers.scala

```diff
@@ -100,6 +100,12 @@ final class Headers private (headers: List[Header])
       this.toList.equals(otherheaders.toList)
     case _ => false
   }
+
+  def redactSensitive(sensitiveHeaderNames: Set[CaseInsensitiveString] = Headers.SensitiveHeaders): Headers =
```

Comment:
This is a good idea.  Could generalize to `CaseInsensitiveString => Boolean`, which is what that Set is.  We did something like this in CORS.